src/opentimelineio/stack.cpp

```diff
@@ -93,4 +93,45 @@ std::vector<SerializableObject::Retainer<Clip>> Stack::clip_if(
     return children_if<Clip>(error_status, search_range, shallow_search);
 }
 
+SerializableObject::Retainer<Bounds> 
+Stack::bounds(ErrorStatus* error_status) const {
+    optional<Imath::Box2d> box;
+    bool found_first_child = false;
+    for (auto child: children()) {
```

Comment:
You could potentially also use the ```children_if()``` function here, like:
```
for (auto clip : children_if<Clip>(error_status))
{
    optional<Imath::Box2d> child_box;
    if (auto child_bounds = clip->bounds(error_status)) {
        child_box = (*child_bounds).box();
    }
    if (*error_status) {
        return Retainer<Bounds>();
    }
    if ( child_box ) {
        if (found_first_child) {
            box->extendBy(*child_box);
        }
        else {
            box = child_box;
            found_first_child = true;
        }          
    }
}
```

Reply:
I like this.  Will do.